### PR TITLE
Make `Iterator.empty` also be `i.Iterable`

### PR DESCRIPTION
**Pull request by NthPortal**
I noticed that `Iterator.empty` is by definition both immutable and repeatedly iterable, so I decided its type may as well reflect that.

If folks dislike changing the typing of the public method, I'm happy to remove the second commit

### CODE REVIEW

**Comment by NthPortal** [comment]
@scala/collections I am both sorry and not sorry to curse you with this

**Comment by lrytz** [comment]
Jenkins `java.io.IOException: No space left on device` is usually handled by @SethTisue - could you write down what you typically delete at https://github.com/scala/scala-jenkins-infra/blob/main/README.md#disk-usage?

File: src/library/scala/collection/Iterator.scala [hunk 1]
```diff
@@ -985,7 +985,7 @@ object Iterator extends IterableFactory[Iterator] {
   override def from[A](source: IterableOnce[A]): Iterator[A] = source.iterator
 
   /** The iterator which produces no values. */
-  @`inline` final def empty[T]: Iterator[T] = _empty
+  @`inline` final def empty[A]: Iterator[A] with immutable.Iterable[A] = _empty
```
Comment (He-Pin) [inline src/library/scala/collection/Iterator.scala:988]: It's a little weird that a method in `Iterator` returns an `Iterable`.

**Comment by NthPortal** [comment]
this change doesn't work because `scanLeft`, `scanRight` and `scan` are expected to return a non-empty `Iterator`, which is *not* repeatedly iterable

**Pull request closed without merging**